docs/api/vulnerability-info.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Vulnerability Info, NuGet API
+description: The data that allows clients to check packages for known vulnerabilities.
+author: zivkan
+ms.author: zivkan
+ms.date: 06/16/2023
+ms.topic: reference
+---
+
+# Vulnerability information
+
+NuGet client, starting from version 6.7, can download known package vulnerability information to use in scenarios such as checking packages during restore operations.
+
+The API consists of at least two files, the [vulnerability index](#vulnerability-index) and one or more [vulnerability data files](#vulnerability-data).
+Known vulnerability data can be partitioned into multiple files, and the vulnerability index provides clients with information needed to cache files, and update the cache, efficiently.
+
+The resource used for building this URL is the `VulnerabilityInfo` resource found in the [service index](service-index.md).
+
+## Versioning
+
+The following `@type` values are used:
+
+@type value                       | Notes
+--------------------------------- | -----
+VulnerabilityInfo/6.7.0 | The initial release
+
+## Suggested partitioning strategy
+
+The files listed in the vulnerability index should ideally be optimized to maximize caching, and therefore minimize updates to large files.
+This will allow clients to minimize the frequency it needs to download updates.
+
+A suggested strategy for vulnerability data partitioning is to have two files, `base.json` and `updates.json`.
+The `base.json` file is updated periodically (for example once a month), and contains all known vulnerabilities at the time the file is regenerated.
+The `updates.json` file should any new advisories published since `base.json` was last regenerated.
+This will allow clients to download the large `base.json` infrequently, while the frequently changing `updates.json` file is always relatively small.
+
+NuGet client does not provide a way for one file to update or remove entries from another file, so if any modifications are needed to any advisory in `base.json`, then `base.json` can be regenerated and all entries in `updates.json` merged.
```

Comment:
You mean client would update the base.json file? I don't understand why client couldn't do this. Can you elaborate on "does not provide a way"?

Reply:
Say `base.json` has an advisory for `GHA-ABC-123`, but that advisory is edited on github after `base.json` is generated. There's no way for `update.json` to tell client "hey, remember `base.json`'s GHA-ABC-123? Delete it, or overwrite it with the contents from this file".

Reply:
It might be worth lightly describing how clients are meant to merge the 1-16 vuln. files. I think it means any vulnerability data present about a given package in the later dated file replaces the earlier dated file. 

This does beg the question of how a vulnerability is redacted. If `base.json` contains "Newtonsoft 13.0.1 is vulnerable", what would `update.json` contain if we want to state "oopsie, Newtonsoft 13.0.1 is not vulnerable after all", i.e. what does a "non-vulnerable" entry look like? I guess the range no longer includes that non-vulnerable version and the range is either shrunk or split into two disjoint ranges?

Reply:
The more I think about it, it seems to only make sense to replace the version ranges per ID as the ID level rather than merging the list of ranges per ID, but again it might be worth calling out in the doc.

Reply:
> I don't understand what this is trying to convey.

Sorry, I see how that was confusing since I had a typo.

If one data file has `"nuget.core": [ "[2.0.0, )" ]` as vulnerable (single range, array with one element) and another has `"nuget.core": [ "[3.0.0, )" ]` as vulnerable (single range, array with one element), the resulting set of vulnerable ranges for `NuGet.Core` would be `[ "[2.0.0, )" ]` or `[ "[3.0.0, )" ] ` depending on the precedence defined by the client (I am guessing timestamp) rather than the resulting set being both `[ "[2.0.0, )", "[3.0.0, )" ]`. That is, _replace_ rather than _merge_.

Reply:
Ok, that makes more sense. In that case, client will see them as two different vulnerabilities, even if they have the same advisory URL.

Since we're communicating asynchronously, I'm just going to add this in case it's not apparent enough already, but this PR is adding public documentation for what is already implemented. This PR is not a design proposal for future feature work.

What do you think about my updated commit?  I tried to explain better that there is no way for one file to "update" or "revoke" a vulnerability defined in another file. All the files are additive, or concatenated, or whatever language you prefer. I'm not sure how to better describe this.

Reply:
The updated commit looks good. It is clear now. 

For other, @zivkan and I chatted over Teams and the data suggests that, although the number of vulnerability changes and removals are higher than expected, it's not something to change now. We can observe other metrics as we ship this version to assess if any further protocol changes are needed.

For reference, there are 342 IDs with vulnerabilities, 24 of which have had at least one vuln. removal and 189 of which had a vulnerability added to an existing version (a lower bound on the "vuln. change" count). In short, this means that vulnerability additions to IDs that already have vulnerabilities or vulnerability changes may indeed be a significant source of data file updates in both the base and the updates blob.